File: examples/github-client/sdks/go/services.go
package sdk

import (
    "context"
)

// GitHubApiService provides access to 11 API endpoints.
type GitHubApiService struct {
    Client *Client
}

// NewGitHubApiService initializes a new GitHubApiService
func NewGitHubApiService(client *Client) *GitHubApiService {
    return &GitHubApiService{Client: client}
}

// GetRepository: GET repos/{owner}/{repo}
func (s *GitHubApiService) GetRepository(ctx context.Context, owner string, repo string) (*ApiResponse, error) {
    pathParams := map[string]string{
        "owner": owner,
        "repo": repo,
    }
    return s.Client.DoRequest(ctx, "GET", "repos/{owner}/{repo}", pathParams, nil, nil, nil)
}

// GetRepositoryCommits: GET repos/{owner}/{repo}/commits
func (s *GitHubApiService) GetRepositoryCommits(ctx context.Context, owner string, repo string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "owner": owner,
        "repo": repo,
    }
    return s.Client.DoRequest(ctx, "GET", "repos/{owner}/{repo}/commits", pathParams, queryParams, nil, nil)
}

// GetRepositoryContributors: GET repos/{owner}/{repo}/contributors
func (s *GitHubApiService) GetRepositoryContributors(ctx context.Context, owner string, repo string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "owner": owner,
        "repo": repo,
    }
    return s.Client.DoRequest(ctx, "GET", "repos/{owner}/{repo}/contributors", pathParams, queryParams, nil, nil)
}

// GetRepositoryIssues: GET repos/{owner}/{repo}/issues
func (s *GitHubApiService) GetRepositoryIssues(ctx context.Context, owner string, repo string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "owner": owner,
        "repo": repo,
    }
    return s.Client.DoRequest(ctx, "GET", "repos/{owner}/{repo}/issues", pathParams, queryParams, nil, nil)
}

// GetRepositoryReadme: GET repos/{owner}/{repo}/readme
func (s *GitHubApiService) GetRepositoryReadme(ctx context.Context, owner string, repo string) (*ApiResponse, error) {
    pathParams := map[string]string{
        "owner": owner,
        "repo": repo,
    }
    return s.Client.DoRequest(ctx, "GET", "repos/{owner}/{repo}/readme", pathParams, nil, nil, nil)
}

// GetUser: GET users/{username}
func (s *GitHubApiService) GetUser(ctx context.Context, username string) (*ApiResponse, error) {
    pathParams := map[string]string{
        "username": username,
    }
    return s.Client.DoRequest(ctx, "GET", "users/{username}", pathParams, nil, nil, nil)
}

// GetUserRepositories: GET users/{username}/repos
func (s *GitHubApiService) GetUserRepositories(ctx context.Context, username string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "username": username,
    }
    return s.Client.DoRequest(ctx, "GET", "users/{username}/repos", pathParams, queryParams, nil, nil)
}

// GetUserStarredRepositories: GET users/{username}/starred
func (s *GitHubApiService) GetUserStarredRepositories(ctx context.Context, username string, queryParams map[string]interface{}) (*ApiResponse, error) {
    pathParams := map[string]string{
        "username": username,
    }
    return s.Client.DoRequest(ctx, "GET", "users/{username}/starred", pathParams, queryParams, nil, nil)
}

// SearchCode: GET search/code
func (s *GitHubApiService) SearchCode(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "search/code", pathParams, queryParams, nil, nil)
}

// SearchRepositories: GET search/repositories
func (s *GitHubApiService) SearchRepositories(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "search/repositories", pathParams, queryParams, nil, nil)
}

// SearchUsers: GET search/users
func (s *GitHubApiService) SearchUsers(ctx context.Context, queryParams map[string]interface{}) (*ApiResponse, error) {
    var pathParams map[string]string = nil
    return s.Client.DoRequest(ctx, "GET", "search/users", pathParams, queryParams, nil, nil)
}

File: examples/github-client/sdks/java/Services.java
package com.app.sdk;

import java.util.HashMap;
import java.util.Map;

public final class Services {
    private Services() {}

    public static class GitHubApiService {
        private final Client client;

        public GitHubApiService(Client client) {
            this.client = client != null ? client : new Client();
        }

        public GitHubApiService() {
            this(new Client());
        }

        /** GET repos/{owner}/{repo} */
        public ApiResponse<String> getRepository(String owner, String repo) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("owner", owner);
            pathParams.put("repo", repo);
            return this.client.send("GET", "repos/{owner}/{repo}", pathParams, null, null, null);
        }

        /** GET repos/{owner}/{repo}/commits */
        public ApiResponse<String> getRepositoryCommits(String owner, String repo, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("owner", owner);
            pathParams.put("repo", repo);
            return this.client.send("GET", "repos/{owner}/{repo}/commits", pathParams, queryParams, null, null);
        }

        /** GET repos/{owner}/{repo}/contributors */
        public ApiResponse<String> getRepositoryContributors(String owner, String repo, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("owner", owner);
            pathParams.put("repo", repo);
            return this.client.send("GET", "repos/{owner}/{repo}/contributors", pathParams, queryParams, null, null);
        }

        /** GET repos/{owner}/{repo}/issues */
        public ApiResponse<String> getRepositoryIssues(String owner, String repo, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("owner", owner);
            pathParams.put("repo", repo);
            return this.client.send("GET", "repos/{owner}/{repo}/issues", pathParams, queryParams, null, null);
        }

        /** GET repos/{owner}/{repo}/readme */
        public ApiResponse<String> getRepositoryReadme(String owner, String repo) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("owner", owner);
            pathParams.put("repo", repo);
            return this.client.send("GET", "repos/{owner}/{repo}/readme", pathParams, null, null, null);
        }

        /** GET users/{username} */
        public ApiResponse<String> getUser(String username) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("username", username);
            return this.client.send("GET", "users/{username}", pathParams, null, null, null);
        }

        /** GET users/{username}/repos */
        public ApiResponse<String> getUserRepositories(String username, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("username", username);
            return this.client.send("GET", "users/{username}/repos", pathParams, queryParams, null, null);
        }

        /** GET users/{username}/starred */
        public ApiResponse<String> getUserStarredRepositories(String username, Map<String, Object> queryParams) {
            Map<String, String> pathParams = new HashMap<>();
            pathParams.put("username", username);
            return this.client.send("GET", "users/{username}/starred", pathParams, queryParams, null, null);
        }

        /** GET search/code */
        public ApiResponse<String> searchCode(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "search/code", pathParams, queryParams, null, null);
        }

        /** GET search/repositories */
        public ApiResponse<String> searchRepositories(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "search/repositories", pathParams, queryParams, null, null);
        }

        /** GET search/users */
        public ApiResponse<String> searchUsers(Map<String, Object> queryParams) {
            Map<String, String> pathParams = null;
            return this.client.send("GET", "search/users", pathParams, queryParams, null, null);
        }

    }

}